### PR TITLE
Remove getCountryCodeSync and handle SecurityErrors better

### PR DESCRIPTION
**Pull request by AshCorr**
## What does this change?

 - Removes `getCountryCodeSync`
 
   From what I can tell this function does almost exactly the same thing as `@guardian/libs` getLocale function, with the only difference being that Sync version doesn't make an API call if it can't find the locale in local storage. I think getCountryCodeSync must be a relic from a bygone era where getLocale could fail in getting the locale from the API call but getLocale now wraps the API call in a try catch block so even if it fails it won't impact execution.

- Handle SecurityErrors better

  I believe whenever a users browser has restrictive privacy settings enabled if an attempt to access localStorage is made a "SecurityError" is thrown. This error isn't very useful to us but it does cause a lot of noise in Sentry.

### CODE REVIEW

File: dotcom-rendering/src/web/lib/getCountryCode.ts [hunk 1]
```diff
@@ -1,46 +1,5 @@
...
-	if (!isString(code)) return false;
-	if (countries.includes(code as CountryCode)) return true;
-	return false;
-};
-
-/*
-	This method can be used as a non async way of getting the country code
-	after getLocaleCode has been called.
- */
-export const getCountryCodeSync = (): CountryCode | null => {
```
Comment (AshCorr) [inline dotcom-rendering/src/web/lib/getCountryCode.ts:30]: All of this is seemingly done by `@guardian/libs` already by https://github.com/guardian/csnx/blob/main/libs/%40guardian/libs/src/locale/getLocale.ts

**Comment by jamesgorrie** [comment]
There's some strangley large increases in some of the bundles - any idea why? e.g. `TimelineAtomWrapper`

**Comment by AshCorr** [comment]
> There's some strangely large increases in some of the bundles - any idea why? e.g. TimelineAtomWrapper

Hm, not sure... if anything this PR should have reduced bundle size, i haven't introduced anything new, just got rid of existing code...

I wonder if its because I started this branch before I went on holidays and I haven't merged main in since so its comparing the state of main of a month ago vs main now (although I thought the `pull_request` event that compressed actions uses runs against an ephermeral merge branch of main and this PRs branch)

**Comment by AshCorr** [comment]
> If the error is triggered via `localStorage` lookups - would it be more clear to colocate the errors handling in the [storage lib](https://github.com/guardian/libs/blob/main/src/storage/storage.ts) instead of having to deal with it every time we try to access it?

Thats a good idea, I wonder if there would be a scenario where you do want to do something with this error though, eg show the user a message like "you need to leave incognito mode to use this feature" which wouldn't be possible if we prevented storage from throwing any security error

File: dotcom-rendering/src/web/lib/getCountryCode.ts (hunk 1 above)
Comment (mxdvl) [reply dotcom-rendering/src/web/lib/getCountryCode.ts:30]: The last place that relied on `getCountryCodeSync` was the Lotame `init` script, which was removed in #3017 :

https://github.com/guardian/dotcom-rendering/pull/3017/files#diff-361662365c3498da8ac478810d45f93c533944db91483c085f2792472fe0d427

**Comment by jamesgorrie** [comment]
> if there would be a scenario where you do want to do something with this error though

Yea, good point - I suppose we can keep suppressing it until we decide we want to do it in 1 place, which we could even do locally with a lib composing this lib.

**Review by OllysCoding — APPROVED** [review]
+1 makes sense to me :)

File: dotcom-rendering/src/web/lib/getCountryCode.ts [hunk 2]
```diff
@@ -49,18 +8,12 @@ export const getCountryCodeSync = (): CountryCode | null => {
...
-			console.log(`Error getting location from libs/getLocale`);
-			window.guardian.modules.sentry.reportError(
-				error,
-				'get-country-code',
-			);
-			locale = getCountryCodeSync();
-			return locale;
-		});
+	return getLocale().catch((error) => {
+		if (error instanceof Error && error.name === 'SecurityError')
```
[the remark this answers is not in the document]
Comment (mxdvl) [reply dotcom-rendering/src/web/lib/getCountryCode.ts:14]: An alternative is to create a helper to like so:

```ts
/** Generally we see SecurityErrors when a users browser has restrictive privacy settings that prevent access to local storage.
 * We should avoid reporting these errors to Sentry as they're not very useful to us and create a lot of noise.
 */
const isSecurityError = (error: unknown) => error instanceof Error && error.name === 'SecurityError';

/* … */

if(isSecurityError(error)) { return null; }
```

**Pull request merged**